lib/archethic/transaction_chain/transaction/cross_validation_stamp.ex

```diff
@@ -44,13 +44,28 @@ defmodule Archethic.TransactionChain.Transaction.CrossValidationStamp do
         cross_stamp = %__MODULE__{inconsistencies: inconsistencies},
         validation_stamp = %ValidationStamp{}
       ) do
-    raw_stamp = ValidationStamp.serialize(validation_stamp)
-
     signature =
-      [raw_stamp, marshal_inconsistencies(inconsistencies)]
-      |> Crypto.sign_with_last_node_key()
+      validation_stamp
+      |> get_row_data_to_sign(inconsistencies)
+      |> Crypto.sign_with_mining_node_key()
 
-    %{cross_stamp | node_public_key: Crypto.last_node_public_key(), signature: signature}
+    %__MODULE__{
+      cross_stamp
+      | node_public_key: Crypto.mining_node_public_key(),
```

Comment:
We should add a `node_mining_key` field in the stamp

Reply:
This is already removed on the next PR